kratos/utilities/parallel_utilities.h

```diff
@@ -93,7 +93,7 @@ class BlockPartition
         {
             for (auto it = mBlockPartition[i]; it != mBlockPartition[i+1]; ++it)
             {
-                f(*it); //note that we pass a reference to the value, not the iterator
+                f(*it); //note that we pass a reference to the function, not the iterator
```

Comment:
I think 'value' was correct here (as in `*it` vs `it`)

Reply:
I thought f was the function you pass to by reference (and not by iterator)

Reply:
I guess you can read it like that as well. Maybe this would be less ambiguous:
```suggestion
                f(*it); //note that we pass a value to the function, not the iterator
```